### ISSUE TITLE
Double nested repeat support

### ISSUE THREAD

**Issue by akuseru** [NotAdam/Lumina#8]
https://github.com/ufx/SaintCoinach/blob/master/SaintCoinach/Definitions/Quest.json#L341-L355

more crazy definitions that are missing.

**Comment by NotAdam** [comment]
Yep, and there can be groups in these or groups of these too, it's a huge mess. Fixing the groups would require them to have a name at least, and then it could be cleaned up a bit but it's pretty disgusting.

I'm not really sure on a good approach for handling these as its all kind of everywhere

**Comment by goaaats** [comment]
How about just parsing it into a fixed-size array, I think that's what SC does

**Comment by NotAdam** [comment]
You can't for groups, because a group might repeat say 20 times and have several members, which would look more like

```
struct BaseParam
{
    uint kind;
    uint value;
} params[6];
```

Which isn't something you can do sequentially as is without sacrificing what the data actually is

**Comment by NotAdam** [comment]
I think it's not too much of a problem if the groups were named, because then you could generate structures for all of them, but as is, it's kind of just meaningless

**Comment by NotAdam** [comment]
Double nested repeat isn't done, nor is crazy shit like this:
```json
{
  "sheet": "GCSupplyDuty",
  "definitions": [
    {
      "type": "repeat",
      "count": 11,
      "definition": {
        "type": "repeat",
        "count": 3,
        "definition": {
          "type": "group",
          "members": [
            {
              "name": "Item",
              "converter": {
                "type": "link",
                "target": "Item"
              }
            },
            {
              "name": "ItemCount"
            }
          ]
        }
      }
    }
  ]
}
```

But the generator has been completely rewritten and correctly supports:
* Arrays
* Groups (needs more work but what's there is a decent start)
* Sheet references, lazily loaded

bitsets are gone and are just read by their columns now instead, easier that way

this is currently on develop, will merge into master and publish to nuget probably tomorrow or the day after once I fix a couple weird things

main thing to note: generated structs have moved to their own nuget package, lumina.generated, potentially changing namespace too but not 100% committed to that yet

**Comment by NotAdam** [comment]
fyi: all of these changes outlined above are available on nuget now

I'm still conflicted on whether double repeat and whatnot is even worth supporting, it's probably going to be more reasonable to manually implement those sheets as required
